openedx/core/djangoapps/user_authn/docs/decisions/0001_simplify_login_ratelimiting.rst

```diff
@@ -0,0 +1,95 @@
+Simplify Login Rate Limiting
+============================
+
+Status
+------
+
+Accepted
+
+Context
+-------
+
+1st party vs 3rd party login
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+edx-platform allows for both 1st party auth, where you provide the LMS with
+your credentials and it gives you back some session tokens, and also 3rd party
+auth, in which you are directed to a 3rd party to authenticate and then
+redirected to the LMS with a token from that third party which is exchanged for
+1st party(LMS) session tokens.
+
+Login User View
+~~~~~~~~~~~~~~~
+
+..
+    TODO: See if I can make this a link instead.
+The ``login_user`` view in ``views/login.py`` is called for both 1st and 3rd
+party login flows.  In the 3rd party login flow, it's called as the callback
+when redirected from the 3rd party login back to the LMS.
+
+Currently this view is accessed through two different endpoints.  It is
+currently a pass-through call for the ``/api/user/v1/account/login_session``
+using a DRF view.  It is also called directly at ``/login_ajax``
+
+Rate limiting Logins
+~~~~~~~~~~~~~~~~~~~~
+
+Currently there are five different rate limiting implementations in use as a
+part of the login flow.
+
+* `django-ratelimit`_ - A 3rd party rate limiting library that allows you to
```

Comment:
Suggestion: Changing this to a [list table](https://docutils.sourceforge.io/docs/ref/rst/directives.html#list-table) would make this quicker to follow.  Possible columns: name (or library), description, 1st party login, 3rd party login.  I'd especially find it nice to have the last two columns in place of all the text to see what applies where.